services/app/src/components/avatarUploadModal/AvatarUploadModal.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Modal } from 'antd';
+import './avatarModal.css';
+import AvatarModalForm from './AvatarModalForm';
+import AvatarModalComplete from './AvatarModalComplete';
+import { useUserContext } from '../../contexts/UserContext';
+import { buildPlayerAvatarUpdateMessage } from '../../api/playerApi';
+import { useWsContext } from '../../contexts/ws-context';
+
+const AvatarUploadModal = ({ showAvatarModal, closeAvatarModal }) => {
+  const { ws } = useWsContext();
+  const userContext = useUserContext();
+ 
+  const [modalStep, setModalStep] = useState(1);
+  const [imageUrl, setImageUrl] = useState('');
+
+  const onAvatarModalCancel = () => {
+    setModalStep(1);
+    closeAvatarModal();
+    setImageUrl('');
+    userContext.setAvatarError(false);
+  }
+
+  const moveToFinalStep = (url) => {
+    try {
+      const urlHref = new URL(url).href;
+      setImageUrl(urlHref);
+      setModalStep(2);
+    } catch {
```

Comment:
Why is this needed? There are not async calls, what can throw here?

Reply:
Without the catch statement, if the ```new URL``` fails to construct a valid object, the error gets printed in the `console.log` but the program does not crash. I thought it shouldn't so I wrapped it in try catch statement.
Removed it.